src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/AdministrationController.php

```diff
@@ -77,27 +102,92 @@ public function indexAction(FormInterface $form = null)
      *
      * @return RedirectResponse
      */
-    public function processFormAction(Request $request)
+    public function processUploadQuotaFormAction(Request $request)
+    {
+        $this->dispatchHook(
+            'actionAdminAdminPreferencesControllerPostProcessUploadQuotaBefore',
+            ['controller' => $this]
+        );
+
+        return $this->processForm(
+            $request,
+            $this->getUploadQuotaFormHandler()
+        );
+    }
+
+    /**
+     * Process the Performance Smarty configuration form.
+     *
+     * @AdminSecurity("is_granted(['update', 'create', 'delete'], request.get('_legacy_controller'))", message="You do not have permission to update this.", redirectRoute="admin_administration")
+     * @DemoRestricted(redirectRoute="admin_administration")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processNotificationsFormAction(Request $request)
+    {
+        $this->dispatchHook(
```

Comment:
@PierreRambaud can't we just have this as some dynamic hook, directly in processForm method? writing it over and over again in every Controllers would be bad

Reply:
This is exactly why this is a work in progress pull request :D Some refactoring are going to be done after the form refactoring.

Reply:
Done :)